src/lib/util/util.ts

```diff
@@ -0,0 +1,294 @@
+				iterateObject(obj[key], path.concat([key]));
+				return;
+			}
+			stringCount++;
+			if (!get(foreignStrings, `${path.join('.')}.${key}`)) missingStrings.push(`${path.join('.')}.${key}`);
+		});
+	}
+	iterateObject(<object> strings);
+	foreignStringCount = stringCount - missingStrings.length;
+	// missing strings
+	if (stringCount > foreignStringCount) return { completion: foreignStringCount / stringCount * 100, missing: missingStrings };
+	return { completion: 100, missing: [] };
+}
+
+/**
+ * Returns an absolute file URL, readable by the ESM loader.
+ * @param baseURL - The base URL of the module.
+ * @param path - The path to the target file.
+ * @returns The absolute file URL.
+ */
+export function getAbsoluteFileURL(baseURL: string, path: string[]): URL {
+	const __dirname = dirname(fileURLToPath(baseURL));
+	return pathToFileURL(resolve(__dirname, ...path));
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function getJSONResponse(body: any): Promise<unknown> {
+	let fullBody = '';
+
+	for await (const d of body) {
+		fullBody += d.toString();
+	}
+	return JSON.parse(fullBody);
+}
+
+/**
+ * Returns a MessageCreateOptions object.
+ * @param inputData - The data to be used. Can be a string, EmbedBuilder, or an array of either.
+ * @param options - Extra data, such as type, components, or files.
+ * @returns The MessageCreateOptions object.
+ */
+export function messageDataBuilder(inputData: string | EmbedBuilder | (string | EmbedBuilder)[], { type = 'neutral', components = null, files = null }: { type?: 'success' | 'neutral' | 'warning' | 'error'; components?: ActionRowBuilder<MessageActionRowComponentBuilder>[]; files?: AttachmentBuilder[]; } = {}): MessageCreateOptions & InteractionReplyOptions {
+	const messageData = Array.isArray(inputData) ? inputData : [inputData];
+	const embedData = messageData.map((msg): EmbedBuilder => {
+		if (typeof msg === 'string') return new EmbedBuilder().setDescription(msg).setColor(<ColorResolvable> colors[type]);
+		if (!msg.data.color) return msg.setColor(<ColorResolvable> colors[type]);
+		return msg;
+	});
+	const opts: MessageCreateOptions & InteractionReplyOptions = { embeds: embedData };
+	if (components !== null) opts.components = components;
+	if (files !== null) opts.files = files;
+	return opts;
+}
+
+export async function settingsPage(interaction: Interaction, guildLocale: string, option: 'language' | 'format'): Promise<{ current: string; embeds: EmbedBuilder[]; actionRow: ActionRowBuilder; }> {
+	let current: string;
+	let embeds: EmbedBuilder[] = [];
+	const actionRow = new ActionRowBuilder<MessageActionRowComponentBuilder>();
+	switch (option) {
+		case 'language':
+			current = `${languageName[guildLocale] ?? 'Unknown'} (${guildLocale})`;
+			actionRow.addComponents(
+				new SelectMenuBuilder()
+					.setCustomId('language')
+					.addOptions(
+						fs.readdirSync(getAbsoluteFileURL(import.meta.url, ['..', '..', 'locales']))
```

Comment:
```js
2022-10-08T13:28:26.231Z [Quaver] INFO: Logging additional output to error.log.

Ava@DESKTOP-5531PRA MINGW64 ~/Documents/GitHub/Quaver (feat/transition-to-typescript)
$ 2022-10-08T13:28:24.967Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Processing command settings
2022-10-08T13:28:24.968Z [Quaver] INFO: [G 906471497231630336 | U 250934589621665793] Executing command settings
2022-10-08T13:28:24.973Z [Discord] ERROR: An error occurred. Quaver will now shut down to prevent any further issues.
2022-10-08T13:28:24.974Z [Quaver] INFO: Shutting down due to discord...
2022-10-08T13:28:24.975Z [Quaver] INFO: Disconnecting from all guilds...
2022-10-08T13:28:24.975Z [Quaver] INFO: [G 906471497231630336] Disconnecting (restarting)
2022-10-08T13:28:26.231Z [Quaver] ERROR: ENOENT: no such file or directory, scandir 'C:\Users\Ava\Documents\GitHub\Quaver\dist\locales'
Error: ENOENT: no such file or directory, scandir 'C:\Users\Ava\Documents\GitHub\Quaver\dist\locales'
    at Module.readdirSync (node:fs:1450:3)
    at settingsPage (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/lib/util/util.js:259:32)
    at Object.execute (file:///C:/Users/Ava/Documents/GitHub/Quaver/dist/commands/settings.js:19:54)
2022-10-08T13:28:26.231Z [Quaver] INFO: Logging additional output to error.log.^C
```

Reply:
whoops forgot to add another line